### DISCUSSION TITLE
Can I find the configuration files that correspond to a pre-built release?

### DISCUSSION THREAD

**Discussion by jglot** [Jyers/Marlin#1762] [Q&A] [answered]
Hi!

I wonder where/if I can find the Configuration*.h files that were used when building the pre-built release binaries. Specifically, I would like to start from the  "E3V2-BLTouch-5x5-HS-v4.2.2-v2.0.1" and make my modifications from there, since I know that it works well on my printer. However, I can't seem to find that version under "Configuration Files/E3V2 Templates/". It's probably reasonable to suppose that it's similar to the "BLTouch-3x3-HighSpeed" configuration, but I'd like to be certain that I have the identical configuration.

Thank you for Myers for a firmware that has been working very well for me!

**Comment by BlueWings172** [comment, marked as answer]
@jglot 

There isn't always all configurations files corresponding to all pre-built releases I guess.

For example, I used to use prebuilt firmware for UBL-BLTouch-**15x15** but I only can find UBL-BLTouch-**10x10** configuration files. So I used the 10x10 as starting point and changed the lines that defines the number of UBL mesh points from 10 to 15.

So in your case, I guess you can do the same thing.

**Comment by BlueWings172** [comment]
@jglot 

I haven't tried this myself (I use UBL), but I think the line you need change in your case is line 1606 in Configuration.h

Change it to:

`#define GRID_MAX_POINTS_X 5`

**Comment by jglot** [comment]
Ok, thank you!

**Comment by scoggins** [comment]
The config files are dynamical generated at compile time using base configs [.github/workflows/compile-configs.yml](https://github.com/Jyers/Marlin/blob/bugfix-JyersUI/.github/workflows/compile-configs.yml) 

What @BlueWings172 said is correct GRID_MAX_POINTS_X is changed from 10 to 5 or 15 depending on the points. The other change that the build script does is set the correct Creality board (v4.2.2 vs v.42.7) and high speed mode.

```yaml
- name: Run PlatformIO BL-Touch
      id: build_configs
      run: |
        if ($HS); then HSTMP=-HS; else HSTMP=; fi;
        if [ "$BOARD" = V4 ]; then BOARDTMP=v4.2.2; else BOARDTMP=v4.2.7; fi;
        echo "::set-output name=filename::E3V2-UBL-BLTouch-${GRID}x${GRID}${HSTMP}-${BOARDTMP}-${VERSION}.bin"
        cp Configuration\ Files/E3V2\ Templates/UBL-BLTouch-10x10/Configuration.h Marlin/Configuration.h
        cp Configuration\ Files/E3V2\ Templates/UBL-BLTouch-10x10/Configuration_adv.h Marlin/Configuration_adv.h
        sed -i "s/#define MOTHERBOARD.*/#define MOTHERBOARD BOARD_CREALITY_$BOARD/g" Marlin/Configuration.h
        sed -i "s/#define GRID_MAX_POINTS_X.*/#define GRID_MAX_POINTS_X $GRID/g" Marlin/Configuration.h
        if ($HS); then TEMP=; else TEMP='\/\/'; fi;
        sed -i "s/[^ ]*#define BLTOUCH_HS_MODE/$TEMP#define BLTOUCH_HS_MODE/g" Marlin/Configuration_adv.h
        pio run
        mkdir temp
        mv .pio/build/STM32F103RET6_creality_maple/*.bin temp/build.bin
      env:
        BOARD: ${{ matrix.board }}
        GRID: ${{ matrix.grid }}
        HS: ${{ matrix.highspeed }}
        VERSION: ${{ env.RELEASE_VERSION }}
```

So if you want a 5 probe UBL with HIGH speed on a v4.2.2 board you'll be setting:

#define MOTHERBOARD BOARD_CREALITY_V4
#define GRID_MAX_POINTS_X 5
#define BLTOUCH_HS_MODE

Also note that if you are compiling up your own version of marlin 2.0  (not jyers) using these settings, that BLTOUCH_HS_MODE has changed to require a boolean so you have to use this for marlin 2.0:

#define BLTOUCH_HS_MODE true

**Reply by jglot** [reply]
That explains a lot, thank you very much!
